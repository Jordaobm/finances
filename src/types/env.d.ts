declare module "@env" {
  export const WEBCLIENTID: string;
}

declare module "*.svg";
declare module "*.png";
declare module "*.jpg";
