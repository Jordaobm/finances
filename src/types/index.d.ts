declare module "*.png";
declare module "*.svg";
declare module "react-native-radial-gradient";
