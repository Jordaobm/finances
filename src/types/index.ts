export interface IconProps {
  color: string;
}

export interface HeaderProps {
  color: "white" | "black";
  onlySettings?: boolean;
}

export interface Operation {
  category: {
    color: string;
  };
  card: {
    name: string;
  };
  value: number;
  date: Date;
  name: string;
}

export interface CardOperationProps {
  operations: Operation[];
}

export interface NavigationProps {
  activeRoute: string;
}
