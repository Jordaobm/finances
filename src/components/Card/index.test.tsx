import { NavigationContainer } from "@react-navigation/native";
import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import { CardComponent, FakeCard } from ".";
import { Card } from "../../types";

jest.mock("react-native-toast-message", () => ({
  show: jest.fn(),
  hide: jest.fn(),
}));

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

let updateCard: Card = {} as Card;

function setUpdateCard(card: Card) {
  updateCard = card;
}

function mockUseUpdateDataContext() {
  return {
    updateCard,
    setUpdateCard,
  };
}

jest.mock("../../context/UpdateDataContext", () => {
  return {
    useUpdateDataContext: mockUseUpdateDataContext,
  };
});

it("must be able to select the wallet", async () => {
  const card: Card = {
    id: "01",
    colorBackground: "#f4f4",
    colorText: "#fff",
    currentValue: 500,
    institutionName: "Carteira",
    name: "Carteira",
    colorBackgroundNumber: 100,
    colorTextNumber: 100,
  };

  const { getByTestId } = await render(
    <NavigationContainer>
      <CardComponent card={card} />
    </NavigationContainer>
  );

  let { updateCard } = mockUseUpdateDataContext();

  await fireEvent(getByTestId("selectWallet"), "onPress", (updateCard = card));
  await expect(updateCard).toEqual(card);
});

it("must be able to select a card", async () => {
  const card: Card = {
    id: `${new Date().getTime()}`,
    colorBackground: "#f4f4",
    colorText: "#fff",
    currentValue: 500,
    institutionName: "Nubank",
    name: "Nubank",
    colorBackgroundNumber: 100,
    colorTextNumber: 100,
  };

  const { getByTestId } = await render(
    <NavigationContainer>
      <CardComponent card={card} />
    </NavigationContainer>
  );

  let { updateCard } = mockUseUpdateDataContext();

  await fireEvent(getByTestId("selectCard"), "onPress", (updateCard = card));
  await expect(updateCard).toEqual(card);
});

it("must be able to select a FakeCard", async () => {
  const { getByTestId } = await render(
    <NavigationContainer>
      <FakeCard />
    </NavigationContainer>
  );

  await fireEvent(getByTestId("selectFakeCard"), "onPress");
});

it("must be able to select a FakeCard", async () => {
  const { getByTestId } = await render(
    <NavigationContainer>
      <FakeCard />
    </NavigationContainer>
  );

  await fireEvent(getByTestId("addCard"), "onPress");
});
