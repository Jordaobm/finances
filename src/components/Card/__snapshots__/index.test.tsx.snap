// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Card for Carteira 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 172,
        "justifyContent": "center",
        "maxWidth": 299,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#f4f4",
        "borderColor": "#ffbf00",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 4,
        "height": 172,
        "justifyContent": "center",
        "maxWidth": 299,
        "opacity": 1,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": "100%",
      }
    }
    testID="walletContent"
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffbf00",
            "borderRadius": 4,
            "height": 4,
            "left": 8,
            "position": "relative",
            "top": -10,
            "width": 50,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffbf00",
            "borderBottomLeftRadius": 10,
            "borderColor": "#ffbf00",
            "borderStyle": "solid",
            "borderTopLeftRadius": 10,
            "borderWidth": 4,
            "height": 50,
            "position": "absolute",
            "right": -3,
            "width": 100,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "Roboto-Bold",
              "fontSize": 28,
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
         R$500,00
      </Text>
    </View>
  </View>
</View>
`;

exports[`Component Card for Fake Card 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 172,
        "justifyContent": "center",
        "maxWidth": 299,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "#c6c6c6",
        "borderRadius": 10,
        "borderStyle": "dashed",
        "borderWidth": 1,
        "height": 172,
        "maxWidth": 299,
        "opacity": 0.4,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": "100%",
      }
    }
    testID="cardContent"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "width": "100%",
          },
          Object {
            "opacity": 0.2,
          },
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="4"
        bbWidth="18"
        fill="none"
        focusable={false}
        height="4"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 4,
              "width": 18,
            },
          ]
        }
        vbHeight={4}
        vbWidth={18}
        width="18"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx="2"
            cy="2"
            fill={4284045657}
            propList={
              Array [
                "fill",
              ]
            }
            r="2"
          />
          <RNSVGCircle
            cx="9"
            cy="2"
            fill={4284045657}
            propList={
              Array [
                "fill",
              ]
            }
            r="2"
          />
          <RNSVGCircle
            cx="16"
            cy="2"
            fill={4284045657}
            propList={
              Array [
                "fill",
              ]
            }
            r="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/chip.png",
        }
      }
      style={
        Array [
          Object {
            "height": 30,
            "marginTop": 40,
            "width": 30,
          },
          Object {
            "opacity": 0.2,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "column",
            "left": "25%",
            "position": "relative",
            "top": "-20%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "Roboto-Bold",
              "fontSize": 28,
            },
            Object {
              "opacity": 0.2,
            },
          ]
        }
      >
        R$ X.XXX,XX
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "Roboto-Bold",
              "fontSize": 14,
              "maxWidth": 190,
            },
            Object {
              "opacity": 0.2,
            },
          ]
        }
      >
        Nome do titular
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "Roboto-Regular",
            "fontSize": 10,
            "left": "10%",
            "position": "absolute",
            "top": "110%",
          },
          Object {
            "opacity": 0.2,
          },
        ]
      }
    >
      Instituição financeira
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "color": "#343434",
        "fontFamily": "Roboto-Bold",
        "fontSize": 18,
        "opacity": 1,
        "position": "absolute",
      }
    }
    testID="addCard"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#343434",
            "fontFamily": "Roboto-Bold",
            "fontSize": 18,
            "opacity": 1,
          },
        ]
      }
    >
      Toque para adicionar um cartão
    </Text>
  </View>
</View>
`;

exports[`Component Card for other cards 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 172,
        "justifyContent": "center",
        "maxWidth": 299,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#f4f4",
        "borderColor": "#c6c6c6",
        "borderRadius": 10,
        "borderStyle": "dashed",
        "borderWidth": 1,
        "height": 172,
        "maxWidth": 299,
        "opacity": 1,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": "100%",
      }
    }
    testID="cardContent"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "width": "100%",
          },
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="4"
        bbWidth="18"
        fill="none"
        focusable={false}
        height="4"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 4,
              "width": 18,
            },
          ]
        }
        vbHeight={4}
        vbWidth={18}
        width="18"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGCircle
            cx="2"
            cy="2"
            fill={4294967295}
            propList={
              Array [
                "fill",
              ]
            }
            r="2"
          />
          <RNSVGCircle
            cx="9"
            cy="2"
            fill={4294967295}
            propList={
              Array [
                "fill",
              ]
            }
            r="2"
          />
          <RNSVGCircle
            cx="16"
            cy="2"
            fill={4294967295}
            propList={
              Array [
                "fill",
              ]
            }
            r="2"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/chip.png",
        }
      }
      style={
        Array [
          Object {
            "height": 30,
            "marginTop": 40,
            "width": 30,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "column",
            "left": "25%",
            "position": "relative",
            "top": "-20%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "Roboto-Bold",
              "fontSize": 28,
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
         R$500,00
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#595959",
              "fontFamily": "Roboto-Bold",
              "fontSize": 14,
              "maxWidth": 190,
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Nubank
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "Roboto-Regular",
            "fontSize": 10,
            "left": "10%",
            "position": "absolute",
            "top": "110%",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Nubank
    </Text>
  </View>
</View>
`;
